scripts/preprocessor.py

```diff
@@ -94,12 +114,22 @@ def preprocess(
 
     gc.collect()
 
-    output_directory = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../datasets/', computation_request_id)
+    datasets_directory = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../datasets/')
+    if (not os.path.exists(datasets_directory)):
+        os.mkdir(datasets_directory)
 
+    output_directory = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../datasets/', computation_request_id)
     if (not os.path.exists(output_directory)):
         os.mkdir(output_directory)
 
```

Comment:
Use `os.makedirs(path, exist_ok=True)` where path is `os.path.join(os.path.dirname(os.path.realpath(__file__)), '../datasets/', computation_request_id)`. There is not need to check twice the existence of the dirs